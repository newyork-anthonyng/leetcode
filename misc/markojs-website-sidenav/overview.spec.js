const myModule = require('./overview');
const { createParentList, createOverview } = myModule;

describe('createOverview', () => {
  it('should create simple overview', () => {
    const structure = [
      {
        title: 'Installation',
        docs: ['First', 'Second', 'Third'],
      },
      {
        title: 'Bundler Integration',
        docs: ['Fourth', 'Fifth', 'Sixth'],
      },
    ];

    expect(createOverview(structure)).toMatchSnapshot();
  });

  it('should create overview with nested overview (none selected)', () => {
    const structure = [
      {
        title: 'Installation',
        docs: ['First', 'Second', 'Third'],
      },
      {
        title: 'Bundler Integration',
        docs: ['Fourth', 'Fifth', 'Sixth',
        { title: 'Other', docs: ['Seventh', 'Eighth'] },
      ],
    },
  ];

  expect(createOverview(structure)).toMatchSnapshot();
});

it('should include ToC for simple overview when selected', () => {
  const structure = [
    {
      title: 'Installation',
      docs: ['First', 'Second', 'Third'],
    },
    {
      title: 'Bundler Integration',
      docs: ['Fourth', 'Fifth', 'Sixth'],
    },
  ];

  expect(createOverview(structure, 'Second', '<h1>Table of Contents</h1>')).toMatchSnapshot();
});

it('should create overview when nested overview is selected', () => {
  const structure = [
    {
      title: 'Installation',
      docs: ['First', 'Second', 'Third'],
    },
    {
      title: 'Bundler Integration',
      docs: ['Fourth', 'Fifth', 'Sixth',
      { title: 'Other', docs: ['Seventh', 'Eighth'] },
    ],
  },
];

expect(createOverview(structure, 'Other')).toMatchSnapshot();
});

it('should create ToC for when child of nested overview is selected', () => {
  const structure = [
    {
      title: 'Installation',
      docs: ['First', 'Second', 'Third'],
    },
    {
      title: 'Bundler Integration',
      docs: ['Fourth', 'Fifth', 'Sixth',
        { title: 'Other', docs: ['Seventh', 'Eighth'] },
      ],
    },
  ];

  expect(createOverview(structure, 'Seventh', '<h1>Table of Contents</h1>')).toEqual();
});
});

describe('createParentList', () => {
  it('should create correct parent list of simple structure', () => {
    const structure = [
      {
        title: 'Installation',
        docs: ['First', 'Second', 'Third'],
      },
      {
        title: 'Bundler Integration',
        docs: ['Fourth', 'Fifth', 'Sixth'],
      },
    ];

    expect(createParentList(structure)).toMatchSnapshot();
  });

  it('should create correct parent list of nested structure', () => {
    const structure = [
      {
        title: 'Installation',
        docs: ['First', 'Second', 'Third'],
      },
      {
        title: 'Bundler Integration',
        docs: ['Fourth', 'Fifth', 'Sixth',
          { title: 'Other', docs: ['Seventh', 'Eighth'] },
        ],
      },
    ];

    expect(createParentList(structure)).toMatchSnapshot();
  });

  it('should handle multiple nested structures', () => {
    const structure = [
      {
        title: 'Installation',
        docs: ['First', 'Second', 'Third'],
      },
      {
        title: 'Bundler Integration',
        docs: ['Fourth', 'Fifth', 'Sixth',
          { title: 'Other', docs: ['Seventh', 'Eighth'] },
          { title: 'Another', docs: ['Ninth', 'Tenth', 'Eleventh'] },
        ],
      },
    ];

    expect(createParentList(structure)).toMatchSnapshot();
  });

  it('should handle deeply nested structures', () => {
    const structure = [
      {
        title: 'Installation',
        docs: ['First', 'Second', 'Third'],
      },
      {
        title: 'Bundler Integration',
        docs: ['Fourth', 'Fifth', 'Sixth',
          { title: 'Other', docs: ['Seventh', 'Eighth',
              { title: 'Deeply Nested', docs: ['Ninth', 'Tenth'] },
            ],
          },
        ],
      },
    ];

    expect(createParentList(structure)).toMatchSnapshot();
  });

  it('should handle multiple deeply nested structures', () => {
    const structure = [
      {
        title: 'Installation',
        docs: ['First', 'Second', 'Third'],
      },
      {
        title: 'Bundler Integration',
        docs: ['Fourth', 'Fifth', 'Sixth',
          { title: 'Other', docs: ['Seventh', 'Eighth',
              { title: 'Deeply Nested', docs: ['Ninth', 'Tenth'] },
            ],
          },
          { title: 'Another', docs: ['Eleventh', 'Twelfth',
              { title: 'Another Deeply Nested', docs: ['Thirteenth', 'Fourteenth'] },
            ],
          },
        ],
      },
    ];

    expect(createParentList(structure)).toMatchSnapshot();
  });
});
