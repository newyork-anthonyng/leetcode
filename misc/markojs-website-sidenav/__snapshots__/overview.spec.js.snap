// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createOverview should create overview when nested overview is selected 1`] = `"<ul><li>Installation</li><li>First</li><li>Second</li><li>Third</li></li></ul><ul><li>Bundler Integration</li><li>Fourth</li><li>Fifth</li><li>Sixth</li><li>Other</li><ul><li>Seventh</li><li>Eighth</li></li></ul></li></ul>"`;

exports[`createOverview should create overview with nested overview (none selected) 1`] = `"<ul><li>Installation</li><li>First</li><li>Second</li><li>Third</li></li></ul><ul><li>Bundler Integration</li><li>Fourth</li><li>Fifth</li><li>Sixth</li><li>Other</li></li></ul>"`;

exports[`createOverview should create simple overview 1`] = `"<ul><li>Installation</li><li>First</li><li>Second</li><li>Third</li></li></ul><ul><li>Bundler Integration</li><li>Fourth</li><li>Fifth</li><li>Sixth</li></li></ul>"`;

exports[`createOverview should include ToC for simple overview when selected 1`] = `"<ul><li>Installation</li><li>First</li><li>Second</li><li><h1>Table of Contents</h1></li><li>Third</li></li></ul><ul><li>Bundler Integration</li><li>Fourth</li><li>Fifth</li><li>Sixth</li></li></ul>"`;

exports[`createParentList should create correct parent list of nested structure 1`] = `
Object {
  "Eighth": Object {
    "Other": true,
  },
  "Fifth": Object {
    "Bundler Integration": true,
  },
  "First": Object {
    "Installation": true,
  },
  "Fourth": Object {
    "Bundler Integration": true,
  },
  "Second": Object {
    "Installation": true,
  },
  "Seventh": Object {
    "Other": true,
  },
  "Sixth": Object {
    "Bundler Integration": true,
  },
  "Third": Object {
    "Installation": true,
  },
}
`;

exports[`createParentList should create correct parent list of simple structure 1`] = `
Object {
  "Fifth": Object {
    "Bundler Integration": true,
  },
  "First": Object {
    "Installation": true,
  },
  "Fourth": Object {
    "Bundler Integration": true,
  },
  "Second": Object {
    "Installation": true,
  },
  "Sixth": Object {
    "Bundler Integration": true,
  },
  "Third": Object {
    "Installation": true,
  },
}
`;

exports[`createParentList should handle deeply nested structures 1`] = `
Object {
  "Eighth": Object {
    "Other": true,
  },
  "Fifth": Object {
    "Bundler Integration": true,
  },
  "First": Object {
    "Installation": true,
  },
  "Fourth": Object {
    "Bundler Integration": true,
  },
  "Ninth": Object {
    "Deeply Nested": true,
    "Other": true,
  },
  "Second": Object {
    "Installation": true,
  },
  "Seventh": Object {
    "Other": true,
  },
  "Sixth": Object {
    "Bundler Integration": true,
  },
  "Tenth": Object {
    "Deeply Nested": true,
    "Other": true,
  },
  "Third": Object {
    "Installation": true,
  },
}
`;

exports[`createParentList should handle multiple deeply nested structures 1`] = `
Object {
  "Eighth": Object {
    "Other": true,
  },
  "Eleventh": Object {
    "Another": true,
  },
  "Fifth": Object {
    "Bundler Integration": true,
  },
  "First": Object {
    "Installation": true,
  },
  "Fourteenth": Object {
    "Another": true,
    "Another Deeply Nested": true,
  },
  "Fourth": Object {
    "Bundler Integration": true,
  },
  "Ninth": Object {
    "Deeply Nested": true,
    "Other": true,
  },
  "Second": Object {
    "Installation": true,
  },
  "Seventh": Object {
    "Other": true,
  },
  "Sixth": Object {
    "Bundler Integration": true,
  },
  "Tenth": Object {
    "Deeply Nested": true,
    "Other": true,
  },
  "Third": Object {
    "Installation": true,
  },
  "Thirteenth": Object {
    "Another": true,
    "Another Deeply Nested": true,
  },
  "Twelfth": Object {
    "Another": true,
  },
}
`;

exports[`createParentList should handle multiple nested structures 1`] = `
Object {
  "Eighth": Object {
    "Other": true,
  },
  "Eleventh": Object {
    "Another": true,
  },
  "Fifth": Object {
    "Bundler Integration": true,
  },
  "First": Object {
    "Installation": true,
  },
  "Fourth": Object {
    "Bundler Integration": true,
  },
  "Ninth": Object {
    "Another": true,
  },
  "Second": Object {
    "Installation": true,
  },
  "Seventh": Object {
    "Other": true,
  },
  "Sixth": Object {
    "Bundler Integration": true,
  },
  "Tenth": Object {
    "Another": true,
  },
  "Third": Object {
    "Installation": true,
  },
}
`;
